db/c_test.c

```diff
@@ -126,6 +126,32 @@ static void CheckGetCF(
   Free(&val);
 }
 
+static void CheckPinGet(rocksdb_t* db, const rocksdb_readoptions_t* options,
+                        const char* key, const char* expected) {
+  char* err = NULL;
+  size_t val_len;
+  const char* val;
+  rocksdb_pinnableslice_t* p;
```

Comment:
so we need a new function rocksdb_create_pinnableslice to create the object like other C API do(e,g rocksdb_writebatch_create), I think it is not necessary here.

Btw, I think it is better to keep the same way with rocksdb_get, return the value and free it manually outside.